core/src/main/scala/kafka/coordinator/group/GroupMetadataManager.scala

```diff
@@ -239,9 +239,13 @@ class GroupMetadataManager(brokerId: Int,
     }
   }
 
+  /**
+   * @return Returning a map of successfully appended topic partitions and a flag indicting whether the HWM has been
+   *         incremented. The caller ought to complete delayed requests for those returned partitions.
+   */
   def storeGroup(group: GroupMetadata,
                  groupAssignment: Map[String, Array[Byte]],
-                 responseCallback: Errors => Unit): Unit = {
+                 responseCallback: Errors => Unit): Map[TopicPartition, LeaderHwChange] = {
```

Comment:
I raised the point before that it's a bit unusual and unintuitive to have both a callback and a return value. Any thoughts on this?

Reply:
The response was https://github.com/apache/kafka/pull/8657#discussion_r452754165

In short, we should have a way of fetching delayed request from partition instead of using return value to carry them.

Reply:
Thanks, I had missed that. Will respond in that thread.